examples/dev_example/docs/build.jl

```diff
@@ -0,0 +1,28 @@
+import Lexicon.Elements: config, section, document, page
+
+import Lexicon.Render: save, markdown
+
+import Docile: Cache, Collector
+
+out = document("TestDocumentation",
+    section("Introduction",
+        page("index", "index.md"),
+        ),
+    section("Docile-API",
+        page("Cache", config(style_obj="###"),
+            "# Cache",
+            "## Macros",
+            "Some intermediate text describing more about macros.",
+            (Cache, config(objfilter=[:macro])),
+            "## Methods",
+            (Cache, config(objfilter=[:method])),
+            ),
+        page("Collector", config(objfilter=[:method, :macro], style_obj="###"),
+            "# Collector",
+            "## Methods & Macros",
+            Collector,
+            ),
+        ),
+    )
```

Comment:
> This is looking much nicer.

Thanks

> Instead of requiring Modules to be separated with tuples for applying different config could we just allow nested section inside of page? The less syntax things to have to remember the better I think.

that was one of the reason I used before a `objs()`. Personlly I would not suggest to allow `sections` within a page: reason: as you said [yourself:](https://github.com/MichaelHatherly/Lexicon.jl/issues/109#issuecomment-102042683) 
```
section would be used to separate the documentation into different parts 
```
**I also use sections for output folder structure** so adding additional section within a page would destroy that feature. Additionally it seems to instroduce weven more unneccesary nesting.

In that case I think it would be better to revert back to something like `objs(Cache; filter=whatever)`

> The first argument to each page node appears to be a title, yes? And then it's kind of repeated again as the third argument. Reasons for this? Do we need the repetition

`document, section, page` first argument is it's name. I use it in error output but more importantly for output folder and file structure. The other e.g. `"# Cache"` is optional and could be anything. So yes I think we need the first node name.